ixmp4/data/api/base.py

```diff
@@ -108,55 +111,98 @@ def get_remote_exception(self, res: httpx.Response, status_code: int):
             )
         return exc.from_dict(json)
 
-    def _request(self, method: str, path: str, *args, **kwargs) -> Mapping | Sequence:
+    def _request(self, method: str, path: str, *args, **kwargs) -> dict | list | None:
         res = self.client.request(method, path, *args, **kwargs)
 
         if res.status_code >= 400:
             raise self.get_remote_exception(res, res.status_code)
         else:
             try:
                 return res.json()
-            except (ValueError, JSONDecodeError):
-                return res.text
-
-    def _get_by_id(self, id: int, *args, **kwargs) -> Mapping[str, Any]:
+            except JSONDecodeError:
+                if res.status_code < 300 and res.text == "":
+                    return None
+            except ValueError:
+                pass
+            raise UnknownApiError(res.text)
+
+    def _get_by_id(self, id: int, *args, **kwargs) -> dict[str, Any]:
         # we can assume this type on create endpoints
         return self._request("GET", self.prefix + str(id) + "/", **kwargs)  # type: ignore
 
-    def _enumerate(self, *args, table: bool = False, **kwargs) -> Mapping | Sequence:
-        json = None
-        params = {}
-        params["table"] = table
-        join_parameters = kwargs.pop("join_parameters", None)
-        join_runs = kwargs.pop("join_runs", None)
-
-        if join_parameters is not None:
-            params["join_parameters"] = join_parameters
-        if join_runs is not None:
-            params["join_runs"] = join_runs
-
-        if self.enumeration_method == "GET":
-            params.update(kwargs)
+    def _enumeration_request(
```

Comment:
I had to think twice before getting that this method actually executes the request and does not just prepare it.
At least for me `_request_enumeration` is easier to understand.